### PR TITLE
FIX-1899 Curve data streaming

### LINKED ISSUE #1899: Open Channel stream for 1 or several timebased curves (streaming window)
Select curves in a TIME log.

Right click → Open – ETP datastream

RIPS will then Open a Data content window similar to the current Right click → Open functionality.

But the difference is that WE will use ETP to setup a streaming of the datasets to the Data Content window. 

The user can start / stop the streaming in the data content window. Data will continously scroll up when the stream is started.

Probably need a timeout after x minutes ?

From Jira WE-652

### PR DESCRIPTION
**Pull request by eliasbruvik**
[//]: # (Request Template Witsml Explorer)

[//]: # (Thank you for contributing to WITSML Explorer! Before submitting this PR, please fill in the following template.)

## Fixes

[//]: # (Write the GitHub issue number starting with #, or, for Equinor only, the Jira issue number starting with WE-)

This pull request fixes #1899, #478 

## Description

Implemented streaming of curve data:
* Added logic in CurveValuesView to periodically look for new data.
    * When turning on streaming mode, the newest data of the log is fetched and replaces any current data.
        * This is done in case the selected range might not cover the end of the log, and in case the log has been updated before enabling stream mode.
        * When the end of the log is fetched, it starts looking for new data.
    * Delay between each call to the API can be changed in the UI.
* Streaming is only enabled for growing logs.
* Streaming works for both the table and the plot.
    * ContentTable shows new rows with a green color that fades back to normal.
    * The plot adds new data rows, but only shows the 300 last rows to avoid lag.
* EditInterval is updated so it can be updated with the current start and end indexes.
    * It is also disabled while streaming.

## Type of change

[//]: # (Mark any of the types of change that apply.)

* [ ] Bugfix
* [x] New feature (non-breaking change which adds functionality)
* [x] Enhancement of existing functionality
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Impacted Areas in Application

[//]: # (List general components of the application that this PR will affect)

* [x] Frontend
* [ ] API
* [ ] WITSML
* [ ] Other (please describe)

## Checklist:

[//]: # (Please tick all the boxes or remove the ones that aren't needed and explain why)

*Communication*
* [ ] I have made corresponding changes to the documentation
* [ ] PR affects application security

*Code quality*
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [ ] New code is covered by passing tests

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: Src/WitsmlExplorer.Frontend/components/ContentViews/EditNumber.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  } = useContext(OperationContext);
+  const [isEdited, setIsEdited] = useState(false);
+  const [value, setValue] = useState<string>(String(defaultValue));
+
+  const submitEditNumber = () => {
+    setIsEdited(false);
+    onSubmit(parseFloat(value) || null);
+  };
+
+  const onInputChange = (e: any) => {
```
Comment (janmarius) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/EditNumber.tsx]: I suggest avoiding the use of the 'any' event type. The [React TypeScript Cheatsheets](https://react-typescript-cheatsheet.netlify.app/) may contain valuable information about available types.

**Review by janmarius — DISMISSED** [review]
I approve the PR, but recommend fixing the event type set to "any".

*Reviewer checklist*
- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

File: Src/WitsmlExplorer.Frontend/components/ContentViews/EditNumber.tsx (hunk 1 above)
Comment (eliasbruvik) [reply Src/WitsmlExplorer.Frontend/components/ContentViews/EditNumber.tsx]: True, and this should probably be done elsewhere in the codebase in the future as well. Changed to `ChangeEvent<HTMLInputElement>`

**Review by janmarius — APPROVED** [review]
*Reviewer checklist*
- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

**Pull request merged**